### PR TITLE
Create filter and slicing operator comparison

### PR DESCRIPTION
**Pull request by seungkyoon**
Created an example in Python and R comparing various related SimpleITK
filters and the Python slicing operator. Also added CMakeLists.txt to
add both examples to testing framework.

### CODE REVIEW

File: Examples/FilterDemonstration/Documentation.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Overview
+--------
+
+There are numerous SimpleITK filters that have similar functions, but very
+important differences. The filters that will be compared are:
+
+ - ``JoinSeriesImageFilter()`` - Joins multiple N-D images into an (N+1)-D  image
+ - ``ComposeImageFilter()`` - Combines several scalar images into a  multicomponent vector image
+ - ``VectorIndexSelectionCastImageFilter()`` - Extracts the selected index of  the input pixel type vector (the input image pixel type must be a vector and  the output a scalar)
+ - ``ExtractImageFilter()`` - Crops an image to the selected region bounds  using vectors; Can also collapse dimensions
```
Comment (blowekamp) [inline Examples/FilterDemonstration/Documentation.rst]: Currently the extract always reduces the image dimension by one unless the dimension is already 2.

File: Examples/FilterDemonstration/FilterDemonstration.R [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+reader <- ImageFileReader()
+reader$SetFileName(args[[1]])
+image_1 <- reader$Execute()
+reader$SetFileName(args[[2]])
+image_2 <- reader$Execute()
+
+# Join the two vector images together
+join <- JoinSeriesImageFilter()
+joined_image <- join$Execute( image_1, image_2 )
+Show(joined_image, "Joined Image", debugOn=TRUE)
```
Comment (blowekamp) [inline Examples/FilterDemonstration/FilterDemonstration.R]: These shows should not be here.

File: Examples/ImageGridManipulation/Documentation.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+Overview
+--------
+
+There are numerous SimpleITK filters that have similar functions, but very
+important differences. The filters that will be compared are:
+
+ - ``JoinSeriesImageFilter()`` - Joins multiple N-D images into an (N+1)-D  image
+ - ``ComposeImageFilter()`` - Combines several scalar images into a  multicomponent vector image
+ - ``VectorIndexSelectionCastImageFilter()`` - Extracts the selected index of  the input pixel type vector (the input image pixel type must be a vector and the output a scalar)
```
Comment (zivy) [inline Examples/ImageGridManipulation/Documentation.rst]: Please add something along the lines: Additionally, this filter can cast the output pixel type (``SetOutputPixelType`` method).

Explains the "Cast" in the filters name.

File: Examples/ImageGridManipulation/Documentation.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ - ``ExtractImageFilter()`` - Crops an image to the selected region bounds  using vectors; Collapses dimensions unless dimension is two
+ - ``CropImageFilter()`` - Similar to ``ExtractImageFilter()``, but crops an  image by an ``itk::Size`` at upper and lower bounds
+ - Image Slicing Operator [only in Python] - Uses pythonic slicing (``img[i:j, k:l]``) to extract a subregion of an image
```
Comment (zivy) [inline Examples/ImageGridManipulation/Documentation.rst]: This also works in R, so just say slicing.

File: Examples/ImageGridManipulation/Documentation.rst [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+(such as RGB).
+
+Extraction Filters
+++++++++++++++++++
+``VectorIndexSelectionCastImageFilter()`` will isolate a single channel in a
+vector image and return a scalar image. On the other hand,
+``ExtractImageFilter()`` and ``CropImageFilter()`` will extract and return a
+subregion of an image, using an ExtractionRegion size and index and
+``itk::Size``'s respectively. However, note that only the
+``ExtractImageFilter()`` collapses dimensions. The Python image slicing
```
Comment (zivy) [inline Examples/ImageGridManipulation/Documentation.rst]: Remove reference to Python as it also works in R, just say "The image slicing..." .

File: Examples/ImageGridManipulation/ImageGridManipulation.R [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+# Two vector images of same pixel type and dimension expected
+reader <- ImageFileReader()
+reader$SetFileName(args[[1]])
+image_1 <- reader$Execute()
+reader$SetFileName(args[[2]])
+image_2 <- reader$Execute()
+
+# Join the two vector images together
```
Comment (zivy) [inline Examples/ImageGridManipulation/ImageGridManipulation.R]: Change comment to something like "Join the two ND vector images to form a ND+1 image."

File: Examples/ImageGridManipulation/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,16 @@
+
+if ( NOT BUILD_TESTING )
+  return ()
+endif ()
+
+if ( NOT SimpleITK_TESTING_NOSHOW )
```
Comment (zivy) [inline Examples/ImageGridManipulation/CMakeLists.txt]: No need for the conditional now that you removed all of the Show commands from the code.

File: Examples/ImageGridManipulation/ImageGridManipulation.R [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+# Two vector images of same pixel type and dimension expected
+reader <- ImageFileReader()
+reader$SetFileName(args[[1]])
+image_1 <- reader$Execute()
+reader$SetFileName(args[[2]])
+image_2 <- reader$Execute()
+
+# Join two N-D Vector image images to form an (N+1)-D image
```
Comment (zivy) [inline Examples/ImageGridManipulation/ImageGridManipulation.R]: repetition "Vector image images", correct to "Vector images..." (also fix in the Python version)

**Comment by blowekamp** [comment]
topic merged into next.

**Comment by zivy** [comment]
When I look at the next branch on github it doesn't seem to have the ImageGridManipulation directory under examples so not seen by read-the-docs. Is the next branch on github synced with itk.org (I do see it there)?

**Comment by blowekamp** [comment]
Looks like there was a mistake with merging this topic. That has been corrected,  and the example looks nicely rendered now! Nice job.
http://simpleitk.readthedocs.io/en/next/Examples/ImageGridManipulation/Documentation.html

**Pull request merged**